File: packages/types/src/tool-description.ts
export interface ToolParameterDescription {
  name: string;
  type: string;
  required: boolean;
  description: string;
  defaultValue?: unknown;
}

export interface ToolDescriptionData {
  id: string;
  type: "STATELESS" | "STATEFUL";
  category?:
    | "filesystem"
    | "shell"
    | "memory"
    | "code"
    | "http"
    | "workflow"
    | "agent"
    | "integration"
    | "interaction";
  description: string;
  parameters: ToolParameterDescription[];
  tips?: string[];
  examples?: string[];
}

File: crates/foundation/wf-types/src/tool_description.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
pub struct ToolParameterDescription {
    pub name: String,
    pub r#type: String,
    pub required: bool,
    pub description: String,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub default_value: Option<serde_json::Value>,
}

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
pub struct ToolDescriptionData {
    pub id: String,
    pub r#type: String,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub category: Option<String>,
    pub description: String,
    pub parameters: Vec<ToolParameterDescription>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub tips: Option<Vec<String>>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub examples: Option<Vec<String>>,
}
